src/Microsoft.AspNetCore.WebHooks.Custom/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.ComponentModel;
+using Microsoft.AspNet.WebHooks.Config;
+using Microsoft.AspNet.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Configuration;
+using System.Linq;
+using Microsoft.AspNet.WebHooks;
+using System;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extension methods for <see cref="IServiceCollection"/>.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public static class ServiceCollectionExtensions
+    {
+        /// <summary>
+        ///  Initializes support for adding custom WebHook support to your ASP.NET project. The functionality 
+        ///  enables users to manage WebHook subscribers, and to send WebHooks to subscribers with matching 
+        ///  registrations.
+        /// </summary>
+        /// <param name="config">The current <see cref="IServiceCollection"/>app.</param>
+        public static void AddCustomWebHooks(this IServiceCollection services, IConfiguration config)
+        {
+            if (config == null)
+            {
+                throw new ArgumentNullException("config");
+            }
+
+            services.AddSingleton<DefaultAssemblyProvider>();
+            services.Configure<SettingsDictionary>(opt =>

```

Comment:
SettingsDictionary seems like an odd thing to configure. Why not WebHookOptions?